### PR TITLE
[Enhancement] Execute immediately after the dynamic partition table is created

### LINKED ISSUE #17195: Refresh the dynamic partition immediately after creating the table

### PR DESCRIPTION
**Pull request by Astralidea**
Signed-off-by: Astralidea <astralidea@163.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17195

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 58 (72.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|36|50|72.00%|[333, 347, 351, 352, 353, 354, 365, 372, 373, 374, 388, 389, 394, 395]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|8|75.00%|[2473, 2474]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -2465,6 +2465,17 @@ private void createOlapOrLakeTable(Database db, CreateTableStmt stmt) throws Ddl
                 colocateTableIndex.removeTable(tableId, olapTable, false /* isReplay */);
             }
         }
+        if (olapTable.getTableProperty().getDynamicPartitionProperty().getEnable()) {
+            new Thread(() -> {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2469]: does dynamic partition scheduler has execution thread pool? better to use the same thread pool and give it a name.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -2465,6 +2465,17 @@ private void createOlapOrLakeTable(Database db, CreateTableStmt stmt) throws Ddl
                 colocateTableIndex.removeTable(tableId, olapTable, false /* isReplay */);
             }
         }
+        if (olapTable.getTableProperty().getDynamicPartitionProperty().getEnable()) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2468]: should check Config.dynamic_partition_enable

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 1 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2469]: dynamic partition scheduler is a background thread. it does not have a thread pool.

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 43 / 58 (74.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|37|50|74.00%|[333, 351, 352, 353, 354, 365, 372, 373, 374, 388, 389, 394, 395]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|8|75.00%|[2473, 2474]|

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by ABingHuang — APPROVED** [review]
LGTM

**Pull request merged**